llvm/lib/Transforms/Scalar/NaryReassociate.cpp

```diff
@@ -559,20 +559,31 @@ NaryReassociatePass::findClosestMatchingDominator(const SCEV *CandidateExpr,
     return nullptr;
 
   auto &Candidates = Pos->second;
+  const auto *FullExpr = SE->getSCEV(Dominatee);
```

Comment:
Why are we using the SCEV of Dominatee here, rather than CandidateExpr? I'm not familiar with this pass, but from a cursory look what we're actually replacing is CandidateExpr, while Dominatee acts as a context instruction.

Reply:
The pass is replacing one of the operands of `Dominatee` with `CandidateInstruction` (which has a SCEV of `CandidateExpr`). What I want to know is if `CandidateInstruction` is "more poisonous" than the original `Dominatee`.

Reply:
But don't you want to know whether it is more poisonous than the operand being replaced, rather than Dominatee?

Reply:
Hmm, I think you're right. Changed it.